core/src/main/java/org/mule/runtime/core/api/construct/Operation.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.api.construct;
+
+import org.mule.runtime.api.component.execution.ExecutableComponent;
+import org.mule.runtime.api.lifecycle.Lifecycle;
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.processor.Processor;
+
+import java.util.List;
+
+/**
+ * A strongly typed operation defined through the Mule language.
+ *
+ * @since 4.5.0
+ */
+public interface Operation extends ExecutableComponent, Lifecycle {
```

Comment:
why would this be api? Who needs this outside of the runtime?

Reply:
For consistency with `Flow` mainly...